### PR TITLE
Bump backport version.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
Backport often fails last time, even if there are no merge conflicts. I hope updating the workflow would fix this.
@VachaShah, could you please describe what was fixed?

### Issues Resolved
Failed ports:
https://github.com/opensearch-project/sql/pull/839#issuecomment-1299300328
https://github.com/opensearch-project/sql/pull/716#issuecomment-1213323984
https://github.com/opensearch-project/sql/pull/745#issuecomment-1213351676
https://github.com/opensearch-project/sql/pull/747#issuecomment-1213364601
https://github.com/opensearch-project/sql/pull/748#issuecomment-1218213039
...

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dai-chen** [comment]
Thanks for working on this. Is there any chance to confirm if the previous failures can be fixed by this indeed? Or it doesn't hurt to upgrade it anyway?

**Comment by Yury-Fridlyand** [comment]
Unfortunately, I can't confirm. I don't know how to reproduce that and I can't test it on a fork.
I hope @VachaShah can give a bit more information - she is the author of CI workflow.

**Comment by VachaShah** [comment]
> Unfortunately, I can't confirm. I don't know how to reproduce that and I can't test it on a fork. I hope @VachaShah can give a bit more information - she is the author of CI workflow.

@Yury-Fridlyand @dai-chen The new version essentially gets latest changes from upstream and adds a new option `filestoskip`. I have not seen issues with the backport workflow failing even though there are no merge conflicts. From the PRs mentioned in the description, I see that it is failing because of merge conflicts.

**Comment by Yury-Fridlyand** [comment]
All backports I listed I did manually and `git` didn't report unresolved conflicts.

**Review by penghuo — APPROVED** [review]

**Comment by VachaShah** [comment]
> All backports I listed I did manually and `git` didn't report unresolved conflicts.

Oh thats weird. I have not seen this issue in any other repos. May be we can keep an eye and see if it happens again?

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -15,14 +15,14 @@ jobs:
...
           app_id: ${{ secrets.APP_ID }}
           private_key: ${{ secrets.APP_PRIVATE_KEY }}
           installation_id: 22958780
 
       - name: Backport
-        uses: VachaShah/backport@v1.1.4
+        uses: VachaShah/backport@v2.1.0
         with:
           github_token: ${{ steps.github_app_token.outputs.token }}
           branch_name: backport/backport-${{ github.event.number }}
```
Comment (VachaShah) [inline .github/workflows/backport.yml:28]: This would be https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/backport.yml#L28

Comment (Yury-Fridlyand) [reply .github/workflows/backport.yml]: Thanks, fixed in ebc91aa00.

**Review by joshuali925 — APPROVED** [review]
i also saw the backport workflow failure but no git conflict for some PRs in the observability repo, seems like an issue in the workflow (i don't see relevant fixes in 2.1.0 vs 1.1.4)

**Review by MaxKsyunz — APPROVED** [review]
This PR is still relevant, right?

**Comment by joshuali925** [comment]
yes, i don't know if this fixes conflict during backport but it's good to use updated versions..

**Pull request merged**